disperser/batcher/batcher.go

```diff
@@ -177,6 +199,136 @@ func (b *Batcher) Start(ctx context.Context) error {
 	return nil
 }
 
+// updateConfirmationInfo updates the confirmation info for each blob in the batch and returns failed blobs to retry.
+func (b *Batcher) updateConfirmationInfo(ctx context.Context, batch *batch, aggSig *core.SignatureAggregation, txnReceipt *types.Receipt) ([]*disperser.BlobMetadata, error) {
+	if txnReceipt.BlockNumber == nil {
+		return nil, fmt.Errorf("HandleSingleBatch: error getting transaction receipt block number")
+	}
+	headerHash, err := batch.BatchHeader.GetBatchHeaderHash()
+	if err != nil {
+		return nil, fmt.Errorf("HandleSingleBatch: error getting batch header hash: %w", err)
+	}
+	batchID, err := b.getBatchID(ctx, txnReceipt)
```

Comment:
This function requests a new receipt if the previous one doesn't contain the logs. Are the logs the only thing that may not be populated? I see that we are using the `TxHash` on L258. Assuming this will always be populated.

Reply:
Yes, `TxHash` will be populated as long as the txn has been published. 
I've seen `logs` missing intermittently in some RPC endpoints when receipt is retrieved too soon